prover/src/provers/agg_prover.rs

```diff
@@ -78,21 +98,46 @@ impl Prover<AggContext> for AggProver {
             .parse::<usize>()
             .unwrap_or_else(|_| panic!("Can not parse {} to usize", force_bits));
         // 3. compress setup
-        setup(
-            &r1_stark.r1cs_file,
-            &r1_stark.pil_file,
-            &r1_stark.const_file,
-            &r1_stark.exec_file,
-            force_bits,
-        )?;
+        if !prove_data_cache.agg_cache.already_cached {
+            log::info!("setup");
+            setup(
+                &r1_stark.r1cs_file,
+                &r1_stark.pil_file,
+                &r1_stark.const_file,
+                &r1_stark.exec_file,
+                force_bits,
+            )?;
+
+            // add r1cs pil, const, exec to cache and update flag
+            let _ = prove_data_cache.add(
+                r1_stark.r1cs_file.clone(),
+                CacheStage::Agg(StarkFileType::R1cs),
+            );
+            let _ = prove_data_cache.add(
+                r1_stark.pil_file.clone(),
+                CacheStage::Agg(StarkFileType::Pil),
+            );
+            let _ = prove_data_cache.add(
+                r1_stark.const_file.clone(),
+                CacheStage::Agg(StarkFileType::Const),
+            );
+            let _ = prove_data_cache.add(
+                r1_stark.exec_file.clone(),
+                CacheStage::Agg(StarkFileType::Exec),
+            );
+            prove_data_cache.update_cache_flag(CacheStage::Agg(StarkFileType::default()));
+        }
 
         // 4. compress exec
         // TODO: place it in StarkProveArgs?
         let wasm_file = format!(
             "{}/{}.recursive1_js/{}.recursive1.wasm",
             r1_circom.output, ctx.task_name, ctx.task_name
         );
+
         log::info!("wasm_file: {}", wasm_file);
+        log::info!("exec");
```

Comment:
Is this logging useful?

Reply:
I think it should be removed too. already done!